shop3.js

```diff
@@ -0,0 +1,67 @@
+const IlpPacket = require('ilp-packet')
+const http = require('http')
+const crypto = require('crypto')
+const Plugin = require('ilp-plugin-payment-channel-framework')
+function base64 (buf) { return buf.toString('base64').replace(/\+/g, '-').replace(/\//g, '_').replace(/=+$/, '') }
+function hash (secret) { return crypto.createHash('sha256').update(secret).digest() }
+function hmac (secret, input) { return crypto.createHmac('sha256', secret).update(input).digest() }
+
+let users = {}
+const store = {}
+const plugin = new Plugin({
+  listener: {
+    port: 9000
+  },
+  incomingSecret: '',
+  maxBalance: '1000000000',
+  prefix: 'g.letter-shop.mytrustline.',
+  info: {
+    currencyScale: 9,
+    currencyCode: 'XRP',
+    prefix: 'g.letter-shop.mytrustline.',
+    connectors: []
+  },
+  _store: {
+    get: (k) => store[k],
+    put: (k, v) => { store[k] = v },
+    del: (k) => delete store[k]
```

Comment:
Might be worth noting in a comment that it wouldn't be a good idea to use an in-memory store like this in practice but this is just for the example